### PR TITLE
Shorten confirmation number displayed to last 7 characters

### LINKED ISSUE #450: Shorten confirmation number displayed to last 7 characters
The confirmation “number” is currently a uuidv4 (a 32 character string of numbers and/or lowercase letters separated with hyphens) and looks like this: 75442486-0878-440c-9db1-a7006c25a39f

The confirmation page asks users to save and record their confirmation numbers. Some users with questions about their retroactive certification may call EDD and provide their confirmation number. It’s difficult to communicate 32 characters over the phone, and potentially have it repeated back to you for confirmation, possibly more than once. 

The goal is to reduce frustration and improve the experience for users and call center representatives by displaying a much shorter confirmation number.

We should **shorten the confirmation number to only display the final seven characters on the confirmation page**, with only capital letters, and no hyphens. The confirmation number listed above should look like this: C25A39F. 
 
The entire 32 characters of the confirmation number are stored in the database, and will continue to be stored in the database. 

Why 7 characters? 
The UIO confirmation number is eight numeric digits. To avoid confusion and potential overlap, the retroactive certification confirmation numbers should ideally be easily distinguishable from the format of any existing EDD confirmation numbers. 

Chances of duplicate confirmation numbers?
Very low. The uuidv4 is hexadecimal, 16^7 is 268 million and we probably won’t have more than a million confirmation codes at the very most (maximum 1.66 milllion). If there's somehow a duplicate, it can be fixed retroactively if needed.

### PR DESCRIPTION
**Pull request by kalvinwang**
This PR shortens the confirmation number displayed to the last 7 characters
image

===

Resolves #450

- [x] Snapshots updated, if output HTML/JS has changed (`npm run test:update-snapshots`)
- [ ] Changes reviewed on mobile using devtools, if output HTML/CSS has changed
- [ ] Spanish checked by adding `?lng=es` to URL e.g. `/guide/benefits?lng=es`, if Spanish updated
- [ ] Automated tests added, if new functionality added
- [ ] Documentation (e.g. READMEs) updated, if relevant
- [ ] IE11 and Edge compatibility confirmed via manual testing in Browserstack, if new libraries added or newish JS/HTML/CSS features used for the first time in this codebase
- [ ] Accessibility & performance audit run using Google Lighthouse, if major changes made

### CODE REVIEW

File: src/client/pages/RetroCertsConfirmationPage/__snapshots__/index.test.js.snap [hunk 1]
```diff
@@ -62,7 +62,7 @@ exports[`<RetroCertsConfirmationPage /> with confirmation number 1`] = `
           t={[Function]}
           values={
             Object {
-              "confirmationNumber": "CONFIRMATION_NUMBER",
+              "confirmationNumber": "_NUMBER",
```
Comment (kalvinwang) [inline src/client/pages/RetroCertsConfirmationPage/__snapshots__/index.test.js.snap:65]: Interesting, I don't know why this snapshot changed in this way...

Comment (idealisms) [reply src/client/pages/RetroCertsConfirmationPage/__snapshots__/index.test.js.snap:65]: Haha, in the test, it sets the confirmation number to CONFIRMATION_NUMBER. Your code is truncating to the last 7 digits :)

File: src/client/pages/RetroCertsConfirmationPage/index.js [hunk 2]
```diff
@@ -25,6 +25,14 @@ function RetroCertsConfirmationPage(props) {
     return <Redirect to={routes.retroCertsWeeksToCertify} />;
   }
 
+  // For historical reasons the confirmation code is stored as a uuidv4
+  // hash in the database, but we display only the last 7 characters to the user.
```
Comment (idealisms) [inline src/client/pages/RetroCertsConfirmationPage/index.js:29]: Can you add to the change description or the github issue how to came up with 7 characters and what the probability of a duplicate with another user is?

Comment (kalvinwang) [reply src/client/pages/RetroCertsConfirmationPage/index.js:29]: OMG, that's hilarious. _NUMBER really doesn't look like a truncated string! Yes I'll add that to the issue now.

Comment (kalvinwang) [reply src/client/pages/RetroCertsConfirmationPage/index.js:29]: Done https://github.com/cagov/unemployment/issues/450

**Pull request merged**